src/api/procedures/redeemTokens.ts

```diff
@@ -47,9 +52,19 @@ export async function prepareRedeemTokens(
     });
   }
 
+  const rawAmount = bigNumberToBalance(amount, context, isDivisible);
+
+  if (from) {
```

Comment:
It seems like `redeemFromPortfolio` accepts a default Portfolio and `portfolioToPortfolioKind` handles defaults as well. 

We should be able to use only `redeemFromPortfolio` and simplify the code here a bit. Like `addInstructionWithMemo` always being used now.

Reply:
That said, if someone had a secondary key permissioned with only the old extrinsic, the way you have it is 100% backward compatible.